source/draw/Canvas.ooc

```diff
@@ -67,6 +67,30 @@ Canvas: abstract class {
 	}
 	fill: abstract func (color: ColorRgba)
 	draw: virtual func ~DrawState (drawState: DrawState) { Debug error("draw~DrawState unimplemented for class " + this class name + "!") }
+	write: virtual func (target, fontAtlas: Image, localOrigin: IntPoint2D, message: Text) {
```

Comment:
The actual implementation that makes testing easier by only taking used settings from DrawState. This function will have settings added when needed instead of overloading. DrawState will handle all alternative data-types and coordinate-systems.